src/Microsoft.AspNetCore.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -57,7 +57,7 @@ public class InputTagHelper : TagHelper
             new Dictionary<string, string>(StringComparer.Ordinal)
             {
                 { "date", "{0:yyyy-MM-dd}" },
-                { "datetime", "{0:yyyy-MM-ddTHH:mm:ss.fffK}" },
+                { "datetime", "{0:yyyy-MM-ddTHH:mm:ss.fff}" },
```

Comment:
Don't change this. The bug is about the `type` attribute values MVC chooses when the `<input>` element has none. If users type in `<input type="datetime" ... />`, we still need to use the appropriate RFC 3339 format.